### DISCUSSION TITLE
Unable to connect records using explicit many-to-many relationship

### DISCUSSION THREAD

**Discussion by momotofu** [prisma/prisma#12525] [Q&A]
👋🏻 Hello,

I'm running into this error `P2018` or `Expected 1 records to be connected after connect operation on one-to-many relation 'PostToPostTag', found 0."`. I've double checked the docs, and I don't get any type errors, but the client crashes when running the code below (see Exhibit A).

Exhibit A: Code failing to succeed
```javascript
const post = await prisma.post.upsert({
    where: {
      id: 1
    },
    create: { name: 'Mock Post' },
    update: { name: 'Mock Post' }
  })

console.log('post: ' post) // A post is successfully logged here

const tag = await prisma.tag.create({
    data: {
       name: 'Mock tag',
       posts: {
         connect: {
           id: post.id // No typescript error here, so it should be fine.
         }
       }
    }
})
```

Schema
```prisma
model Post {
  id     String         @id @default(cuid())
  title  String
  tags   PostTag[]
}

model PostTag {
  id      String    @id @default(cuid())
  tag     Tag?      @relation(fields: [tagId], references: [id], onDelete: Cascade)
  post    Post?     @relation(fields: [postId], references: [id], onDelete: Cascade)
  tagId   String?
  postrId String?
}

model Tag {
  id    String          @id
  name  String
  posts PostTag[]
}
```

Error logged to the console:
```
test = await prisma.post.create(
  The required connected records were not found. Expected 1 records to be connected after connect operation on one-to-many relation 'PostToPostTag', found 0.
details:
{
  code: 'P2018',
  clientVersion: '3.11.0',
  meta: {
    details: "Expected 1 records to be connected after connect operation on one-to-many relation 'PostToPostTag', found 0."
  }
}
```

**Comment by austincrim** [comment]
Hey @momotofu 👋 ,

Since you are using an **explicit** many-to-many relation, the nested `connect` call is actually looking for a `PostTag` with an `id` of `post.id`. The `posts` field on the `tag` model is of type `PostTag`.

If you used an [**implicit** many-to-many](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations), your code should work how you are expecting. Otherwise, you will need to explicitly create the `PostTag` or use a [`connectOrCreate`](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) in your create `tag` call.

**Comment by betofigueiredo** [comment]
@momotofu you need to create the relation first (PostTag) and then connect:

```javascript
const tag = await prisma.tag.create({
  data: {
    name: 'Mock tag',
    posts: {
      create: [
        {
          post: {
            connect: { id: post.id },
          },
        },
      ],
    }
  }
});
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
